docs/lifecycle.rst

```diff
@@ -0,0 +1,183 @@
+**Source**: this document is a shorter and updated version of the
```

Comment:
Could you add an overall title for this page? That will look a bit nicer in the table of contents.

For example:

```suggestion
=========================
Data Dictionary lifecycle
=========================

**Source**: this document is a shorter and updated version of the
```